pkg/splunk/controller/configmap.go

```diff
@@ -49,3 +49,22 @@ func ApplyConfigMap(client splcommon.ControllerClient, configMap *corev1.ConfigM
 
 	return err
 }
+
+// getConfigMap gets the ConfigMap resource in a given namespace
+func getConfigMap(client splcommon.ControllerClient, namespacedName types.NamespacedName) (corev1.ConfigMap, error) {
+	var configMap corev1.ConfigMap
+	err := client.Get(context.TODO(), namespacedName, &configMap)
+	if err != nil {
+		return configMap, err
+	}
+	return configMap, nil
+}
+
+// GetConfigMapResourceVersion gets the Resource version of a configMap
+func GetConfigMapResourceVersion(client splcommon.ControllerClient, namespacedName types.NamespacedName) (string, error) {
```

Comment:
Can we make this an API to retrieve a specific field such as ResourceVersion, UID etc... based on a stringed parameter? If possible, that api can be used for any runtime object as we are using the namespaced name as the parameter for client.Get() anyways. Seems expensive to have an API just for resource version. We can take this up in the future as well :)

Reply:
Its always good to have APIs for metadata. But, as we found that K8 already has the APIs, i guess, we don't have any reason to implement and maintain.

Reply:
Arjun, in the case of run-time objects, we should be still able to use the K8 APIs. Do you see any issues? I think we don't need to implement redundantly anything that is already there in K8 APIs, as it becomes redundant and additional maintenance(?).

Reply:
If we already have the runtime object with us retrieved then we should use the inbuilt apis for sure. My point was if we are retrieving any runtime object as well extracting a particular field from it we can have a generic API for that so as to avoid having a get() API call in the calling code which might look unpleasant.